src/background.js

```diff
@@ -0,0 +1,38 @@
+function runShowMeTheCode(tabId, tab) {
+    function isGitHub(tab) {
+        return urlMatches(tab, /https:\/\/github\.com\/.*/)
+    };
+
+    function isBlobView(tab) {
+        return urlMatches(tab, /.*\/blob\/.*/);
+    };
+
+    function pullRequestFileView(tab) {
+        return urlMatches(tab, /.*\/pull\/.*\/files/);
+    };
+
+    function urlMatches(tab, expression) {
+        return tab.url && tab.url.match(expression);
```

Comment:
Chrome executes this code on all pages.
18 gru 2015 6:07 PM "Łukasz Nowak" <notifications@github.com> napisał(a):

> In src/background.js
> <https://github.com/Viridovix/show-me-the-code-github/pull/32#discussion_r48046642>
> :
>
> > @@ -0,0 +1,38 @@
> > +function runShowMeTheCode(tabId, tab) {
> > +    function isGitHub(tab) {
> > +        return urlMatches(tab, /https:\/\/github\.com\/.*/)
> > +    };
> > +
> > +    function isBlobView(tab) {
> > +        return urlMatches(tab, /.*\/blob\/.*/);
> > +    };
> > +
> > +    function pullRequestFileView(tab) {
> > +        return urlMatches(tab, /.*\/pull\/.*\/files/);
> > +    };
> > +
> > +    function urlMatches(tab, expression) {
> > +        return tab.url && tab.url.match(expression);
>
> Why do we need tab.url check?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Viridovix/show-me-the-code-github/pull/32/files#r48046642>
> .
>